--- ./configure.ac~	2021-05-29 10:48:15.000000000 +0100
+++ ./configure.ac	2021-05-29 10:52:08.857875517 +0100
@@ -27,12 +27,12 @@
 AC_ARG_ENABLE(debug,
           AC_HELP_STRING([--enable-debug], [build debug executable]))
 if test "$enable_debug" = "yes"; then
-   CFLAGS="$CFLAGS -Wall -O0 -g -D_DEBUG -fcommon"
-   CXXFLAGS="$CXXFLAGS -Wall -O0 -g -D_DEBUG -fcommon"
+   AM_CFLAGS="$CFLAGS -Wall -O0 -g -D_DEBUG -fcommon"
+   AM_CXXFLAGS="$CXXFLAGS -Wall -O0 -g -D_DEBUG -fcommon"
    AC_MSG_RESULT([yes])
 else
-   CFLAGS="$CFLAGS -O3 -fcommon"
-   CXXFLAGS="$CXXFLAGS -O3 -fcommon"
+   AM_CFLAGS="$CFLAGS -fcommon"
+   AM_CXXFLAGS="$CXXFLAGS -fcommon"
    AC_MSG_RESULT([no])
 fi
 
