--- ./src/hostfs.c.orig	2021-04-22 14:27:26.000000000 +0100
+++ ./src/hostfs.c	2021-04-22 15:36:11.771255704 +0100
@@ -138,7 +138,7 @@
 static HostFSState hostfs_state = HOSTFS_STATE_UNREGISTERED;
 
 #ifdef NDEBUG
-static inline void dbug_hostfs(const char *format, ...) { NOT_USED(format); }
+static inline void dbug_hostfs(const char *format __attribute__ ((unused)), ...) {}
 #else
 static void
 dbug_hostfs(const char *format, ...)
